### ISSUE TITLE
Replay not handled correctly when having a pause in the middle of a play

### ISSUE THREAD

**Issue by PercyDan54** [ppy/osu-stable-issues#574]
I paused here when I was playing it. And I didn't got a combobreak after I resumed.
But the replay's combo broke here. Seems it counted the click I clicked the orange resume circle (don't know what it's called), so it judges as too early.
I don't have other proof that I didn't get a combo break except the combo record in the replay file was 606 but it broke at 273.
Version replay was made: b20201017.1cuttingedge
In cuttingedge: combo broke
![screenshot002](https://user-images.githubusercontent.com/50285552/96333912-59f4ec00-109f-11eb-8077-64042caf1a73.jpg)
In stable fallback also broke:
![screenshot005](https://user-images.githubusercontent.com/50285552/96333931-7e50c880-109f-11eb-8346-a26590ba1529.jpg)
Replay:
[PercyDan - t+pazolite - Oshama Scramble! [Shizuqua's ADVANCED] (2020-10-17) Osu.zip](https://github.com/ppy/osu-stable-issues/files/5395329/PercyDan.-.t%2Bpazolite.-.Oshama.Scramble.Shizuqua.s.ADVANCED.2020-10-17.Osu.zip)

**Comment by Magnus-Cosmos** [comment]
This is because pressing the orange circle counts as a keypress in the replay (but not while playing).

**Comment by PercyDan54** [comment]
> 
> 
> This is because pressing the orange circle counts as a keypress in the replay (but not while playing).

I think this is a bug and should be fixed

**Comment by peppy** [comment]
I've added a probably fix for this on the latest cutting edge release. Could you please try and reproduce the same issue and let me know if it helps?

**Comment by PercyDan54** [comment]
> 
> 
> I've added a probably fix for this on the latest cutting edge release. Could you please try and reproduce the same issue and let me know if it helps?

Sorry but maybe it's too hard since the resume circle's position is random every time.
Could you send me a custom version which resume circle's position will always be on the next hitobject?

**Comment by peppy** [comment]
just make a quick beatmap in the editor with huge CS?

**Comment by PercyDan54** [comment]
> 
> 
> just make a quick beatmap in the editor with huge CS?

No.
There's still one keypress recorded in the replay.
But I didn't press any keys except once resuming the game
[![BwvuQS.png](https://s1.ax1x.com/2020/11/01/BwvuQS.png)](https://imgchr.com/i/BwvuQS)

**Comment by peppy** [comment]
Can you confirm you're on 20201101?

**Comment by PercyDan54** [comment]
> 
> 
> Can you confirm you're on 20201101?

Yes. I went to the settings to check for update as soon as I recieved your replay, and it says there's a new version.
And I restarted osu after it downloaded

**Comment by PercyDan54** [comment]
> 
> 
> Can you confirm you're on 20201101?

Since you have the source code and can change code without restarting, here's the step I used to reproduce :
1. Create a short beatmap with HP 0 with some hit objects.
2. Go to the map, dont press any key, pause and resume once.
3.View the replay to check if there's any press recorded, if there is, this issue probably remains

**Comment by peppy** [comment]
I'm not sure this can be amicably fixed in stable, but for the benefit of lazer, what is the expectation here? Currently in lazer, the click *does* count as a gameplay click, which may make more sense?

**Comment by PercyDan54** [comment]
> I'm not sure this can be amicably fixed in stable, but for the benefit of lazer, what is the expectation here? Currently in lazer, the click _does_ count as a gameplay click, which may make more sense?

Either:
Don't record the resume click
Make it count as a gameplay click
